File: python/preliminar/src/app/modules/auth/dtos/outputs/login_auth_managers_dto.py
from pydantic import BaseModel, ConfigDict, field_serializer
from datetime import datetime

from app.modules.managers.entity import ManagerEntity

class LoginAuthManagersOutputDto(BaseModel):
    token: str
    usuario: ManagerEntity
    expiraEm: datetime

    model_config = ConfigDict(serialization={'datetime_format': 'iso8601'})

    def to_json(self):
        return self.model_dump()

    @field_serializer("usuario")
    def serialize_usuario(self, usuario: ManagerEntity):
        return usuario.to_json()

    @field_serializer("expiraEm")
    def serialize_expira_em(self, value: datetime):
        """Garante que o datetime seja convertido para ISO 8601"""
        return value.isoformat()

File: typescript/express_orm/src/app/modules/auth/dtos/outputs/login-auth-managers.dto.ts
import { Transform, instanceToPlain } from "class-transformer";

// Entities
import { ManagerEntity } from "@/app/modules/managers/entities/manager.entity";

export class LoginAuthManagersOutputDto {
  @Transform(({ value }) => value.toJSON())
  usuario: Partial<ManagerEntity>;

  token: string;
  expiraEm: string; // seconds

  constructor(partial: Partial<LoginAuthManagersOutputDto>) {
    Object.assign(this, partial);
  }

  toJSON() {
    return instanceToPlain(this);
  }
}
